### PR TITLE
Gh429

### PR DESCRIPTION
**Pull request by MarkMLunarG**
tests: GH429 Addition of tests for shader_checker validation

This branch adds tests that challenge the validation layer on
various shader checker cases including:

1) BAD_SPECIALIZATION
    Generic vkCreateGraphisPipe test
    CreatePipelineCheckShaderBadSpecialization
2) PUSH_CONSTANT_NOT_ACCESSIBLE_FROM_STAGE
    Generic vkCreateGraphisPipe test
     CreatePipelineCheckShaderPushConstantNotAccessible
3) DESCRIPTOR_NOT_ACCESSIBLE_FROM_STAGE
    Generic vkCreateGraphisPipe test
    CreatePipelineCheckShaderDescriptorNotAccessible
4) DESCRIPTOR_TYPE_MISMATCH
    Generic vkCreateGraphisPipe test
    CreatePipelineCheckShaderDescriptorTypeMismatch
5) FEATURE_NOT_ENABLED
    Generic vkCreateGraphisPipe test
    CreatePipelineCheckShaderNotEnabled
6) BAD_CAPABILITY
    Generic vkCreateGraphisPipe test
    CreatePipelineCheckShaderBadCapability

docs: GH429 change doc for shader_checker validation tests

### CODE REVIEW

**Comment by tobine** [comment]
This LGTM but would like to see a thumbs up from @chrisforbes before the push as he's the expert on these errors.
Passes on our farm and no unexpected errors for new tests.

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -13360,6 +13360,505 @@ TEST_F(VkLayerTest, CreatePipelineVertexOutputNotConsumed) {
...
+    pipe.CreateVKPipeline(pipeline_layout, renderPass());
+    m_errorMonitor->VerifyFound();
+
+    vkDestroyPipelineLayout(m_device->device(), pipeline_layout, nullptr);
+    vkDestroyDescriptorPool(m_device->device(), descriptorset_pool, nullptr);
+    vkDestroyDescriptorSetLayout(m_device->device(), descriptorset_layout, nullptr);
+}
+
+TEST_F(VkLayerTest, CreatePipelineCheckShaderDescriptorNotAccessible) {
+    TEST_DESCRIPTION("Challenge core_validation with shader validation issues related to vkCreateGraphicsPipelines.");
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: These descriptions could be better.

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -13360,6 +13360,505 @@ TEST_F(VkLayerTest, CreatePipelineVertexOutputNotConsumed) {
...
+    buffer_info.range = 1024;
+
+    VkWriteDescriptorSet write_descriptor_set = {};
+    write_descriptor_set.sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+    write_descriptor_set.dstSet = descriptorset;
+    write_descriptor_set.descriptorCount = 1;
+    write_descriptor_set.descriptorType = VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER;
+    write_descriptor_set.pBufferInfo = &buffer_info;
+
+    vkUpdateDescriptorSets(m_device->device(), 1, &write_descriptor_set, 0, nullptr);
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: You don't need an actual descriptor set to test this -- just the layout.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -13360,6 +13360,505 @@ TEST_F(VkLayerTest, CreatePipelineVertexOutputNotConsumed) {
...
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: Same comment as next test -- this can be slimmed down a lot.

File: tests/layer_validation_tests.cpp [hunk 4]
```diff
@@ -13360,6 +13360,505 @@ TEST_F(VkLayerTest, CreatePipelineVertexOutputNotConsumed) {
...
+        // id,  offset,                size
+        {0, 4, sizeof(uint32_t)};  // Challenge core validation by using a bogus offset.
+
+    uint32_t data = 1;
+
+    // Set up the info describing spec map and data
+    const VkSpecializationInfo specialization_info = {
+        1,                 // mapEntryCount
+        &entry,           // pMapEntries
+        1 * sizeof(float), // dataSize
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: nitpick, but it would be nice if the types were consistent.

Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: I think this test does the right thing :)

File: tests/layer_validation_tests.cpp [hunk 5]
```diff
@@ -13360,6 +13360,505 @@ TEST_F(VkLayerTest, CreatePipelineVertexOutputNotConsumed) {
...
+
+    VkPipelineColorBlendStateCreateInfo color_blend_state_create_info = {};
+    color_blend_state_create_info.sType = VK_STRUCTURE_TYPE_PIPELINE_COLOR_BLEND_STATE_CREATE_INFO;
+    color_blend_state_create_info.attachmentCount = 1;
+    color_blend_state_create_info.pAttachments = &color_blend_attachment_state;
+
+    VkGraphicsPipelineCreateInfo graphicspipe_create_info = {};
+    graphicspipe_create_info.sType = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
+    graphicspipe_create_info.stageCount = 2;
+    graphicspipe_create_info.pStages = shader_stage_create_info;
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: A shame to have to write all of this noise, but we can slim it down later with framework fixes...

File: tests/layer_validation_tests.cpp (hunk 4 above)
Comment (MarkMLunarG) [inline tests/layer_validation_tests.cpp]: Nice work. That was a just a test.

**Comment by chrisforbes** [comment]
Any movement on this? Would be good to get it landed before it bitrots.

**Comment by mark-lunarg** [comment]
Yeah, sorry @chrisforbes .  I hope to get to it in the next couple-three days.  If you have the time and inclination please feel free to make you suggested changes and push it.

**Comment by mark-lunarg** [comment]
Updated for review feedback, fixed unexpected errors and pushed to Master.

**Pull request closed without merging**